GVFS/GVFS.Common/GVFSEnlistmentStatistics.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public class GVFSEnlistmentStatistics
+    {
+        private List<string> gitPaths;
+        private List<string> placeholderFolderPaths;
+        private List<string> placeholderFilePaths;
+        private List<string> modifiedFolderPaths;
+        private List<string> modifiedFilePaths;
+        private int trackedFilesCount;
+        private int placeholderCount;
+        private int modifiedPathsCount;
+        private Dictionary<string, int> trackedFilesDirectoryTracking;
+        private Dictionary<string, int> hydratedFilesDirectoryTracking;
+
+        public GVFSEnlistmentStatistics(List<string> gitPaths, List<string> placeholderFolderPaths, List<string> placeholderFilePaths, List<string> modifiedFolderPaths, List<string> modifiedFilePaths)
+        {
+            this.gitPaths = gitPaths;
+            this.placeholderFolderPaths = placeholderFolderPaths;
+            this.placeholderFilePaths = placeholderFilePaths;
+            this.modifiedFolderPaths = modifiedFolderPaths;
+            this.modifiedFilePaths = modifiedFilePaths;
+        }
+
+        public void CalculateStatistics(string parentDirectory)
+        {
+            this.trackedFilesCount = 0;
+            this.placeholderCount = 0;
+            this.modifiedPathsCount = 0;
+            this.trackedFilesDirectoryTracking = new Dictionary<string, int>();
+            this.hydratedFilesDirectoryTracking = new Dictionary<string, int>();
+
+            this.CategorizePaths(this.gitPaths, this.trackedFilesDirectoryTracking, ref this.trackedFilesCount, parentDirectory, isFile: true);
+            this.CategorizePaths(this.placeholderFolderPaths, this.hydratedFilesDirectoryTracking, ref this.placeholderCount, parentDirectory, isFile: false);
+            this.CategorizePaths(this.placeholderFilePaths, this.hydratedFilesDirectoryTracking, ref this.placeholderCount, parentDirectory, isFile: true);
+            this.CategorizePaths(this.modifiedFolderPaths, this.hydratedFilesDirectoryTracking, ref this.modifiedPathsCount, parentDirectory, isFile: false);
+            this.CategorizePaths(this.modifiedFilePaths, this.hydratedFilesDirectoryTracking, ref this.modifiedPathsCount, parentDirectory, isFile: true);
+        }
+
+        public List<string> GetDirectoriesSortedByHydration()
+        {
+            List<string> directoriesSortedByHydration = new List<string>();
+
+            foreach (KeyValuePair<string, int> keyValuePair in this.trackedFilesDirectoryTracking)
+            {
+                directoriesSortedByHydration.Add(keyValuePair.Key);
+            }
+
+            directoriesSortedByHydration.Sort(
+                delegate(string directoryOne, string directoryTwo)
+                {
+                    return this.GetHydrationOfDirectory(directoryOne).CompareTo(this.GetHydrationOfDirectory(directoryTwo)) * -1; // Multiply by -1 to reverse the sort

```

Comment:
Instead of doing '* -1' would` directoriesSortedByHdyrate.SortDecencending` word?

Reply:
I can't find any `SortDescending` method in the API (https://www.geeksforgeeks.org/different-ways-to-sort-an-array-in-descending-order-in-c-sharp/). It looks like there's something in Linq? But I'm not quite certain what that is. Instead, I swapped the order of directoryOne and directoryTwo:
`return this.GetHydrationOfDirectory(directoryTwo).CompareTo(this.GetHydrationOfDirectory(directoryOne));`
Which reverses the order without a `-1`. Also added a comment to make sure it was clear what I was doing

Reply:
I was thinking Linq  OrderByDescending

https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.orderbydescending?view=netframework-4.8

Reply:
OK yeah I was able to get that working 👍